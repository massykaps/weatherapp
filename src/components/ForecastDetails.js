import React from "react";
import PropTypes from "prop-types";
import "../styles/ForecastDetails.css";

const ForecastDetails = ({ forecast }) => {
  const { date, temperature, wind, humidity } = forecast;
  const formattedDate = new Date(date).toDateString();
  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__date">{formattedDate}</div>
      <div className="forecast-details__tempMin">
        Min Temp: {temperature.min}&deg;C
      </div>
      <div className="forecast-details__tempMax">
        Max Temp: {temperature.max}&deg;C
      </div>
      <div className="forecast-details__windSpeed">
        Wind Speed: {wind.speed}mph
      </div>
      <div className="forecast-details__windDirection">
        Wind Direction: {wind.direction}
      </div>
      <div className="forecast-details__humidity">Humidity: {humidity}%</div>
    </div>
  );
};

export default ForecastDetails;

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    temperature: PropTypes.shape({
      min: PropTypes.number,
      max: PropTypes.number,
    }).isRequired,
    humidity: PropTypes.number.isRequired,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }).isRequired,
  }).isRequired,
};
