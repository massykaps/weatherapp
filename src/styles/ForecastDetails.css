.forecast-details {
  text-align: center;
  margin-top: 10px;
}

.forecast-details__date {
  font-weight: bold;
  font-size: 20px;
}
