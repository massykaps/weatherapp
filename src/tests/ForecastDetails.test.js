import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    temperature: {
      min: 3,
      max: 13,
    },
    wind: {
      speed: 30,
      direction: "s",
    },
    humidity: 80,
  };

  it("renders", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
