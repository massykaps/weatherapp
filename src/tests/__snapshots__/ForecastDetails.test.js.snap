// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-details__tempMin"
    >
      Min Temp: 3°C
    </div>
    <div
      class="forecast-details__tempMax"
    >
      Max Temp: 13°C
    </div>
    <div
      class="forecast-details__windSpeed"
    >
      Wind Speed: 30mph
    </div>
    <div
      class="forecast-details__windDirection"
    >
      Wind Direction: s
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity: 80%
    </div>
  </div>
</DocumentFragment>
`;
